astropy/coordinates/baseframe.py

```diff
@@ -1201,6 +1201,11 @@ def transform_to(self, new_frame):
                                       'please comment at https://github.com/astropy/astropy/issues/6280')
 
         if inspect.isclass(new_frame):
+            warnings.warn("Transforming a frame instance to a frame class (as opposed to another "
+                          "frame instance) will not be supported in the future.  Either "
+                          "explicitly instantiate the target frame or first convert this frame "
```

Comment:
"this frame instance" is perhaps a little unclear. How about "or first convert the source frame instance to a ...SkyCoord and its its transform_to method."

Reply:
Done